cpp/src/gandiva/precompiled/time.cc

```diff
@@ -944,6 +945,92 @@ gdv_int64 castBIGINT_daytimeinterval(gdv_day_time_interval in) {
          extractDay_daytimeinterval(in) * MILLIS_IN_DAY;
 }
 
+static const char* PATTERN_MONTH[] = {"MONTH", "MON", "MM"};
+static const int PARTTERN_MONTH_LEN[] = {5, 3, 2};
+
+static const char* PATTERN_YEAR[] = {"YEAR", "YYYY", "YY"};
+static const int PARTTERN_YEAR_LEN[] = {4, 4, 2};
+
+FORCE_INLINE
+const char* date_trunc_timestamp_utf8(int64_t context, gdv_int64 date,
+                                      const char* pattern_name, gdv_int32 pattern_length,
+                                      gdv_int32* out_len) {
```

Comment:
You should use `gdv_TYPE``only when you are creating macros. Use the type name in other cases for readability:
```
gdv_int64 -> int64_t
```